src/utils/aws.py

```diff
@@ -99,6 +99,9 @@ def upload_logs_to_cloudwatch(
                         }
                     )
 
+            # Sort log events by timestamp
+            log_events.sort(key=lambda event: event["timestamp"])
```

Comment:
With both Spark and Python writing to the same log file, we sometimes get timestamps that are slightly out of order. When that happens, CloudWatch throws an error on upload. This orders all log records to prevent that.